lib/ide/src/view/text_editor.rs

```diff
@@ -45,54 +88,65 @@ impl TextEditor {
         let text_size    = 16.0;
         let properties   = TextFieldProperties {font,text_size,base_color,size};
         let text_field   = TextField::new(&world,properties);
+        let key_listener = None;
+        let controller_clone = controller.clone_ref();
+        let text_field_clone = text_field.clone();
+        executor::global::spawn(async move {
+            if let Ok(content) = controller_clone.read_content().await {
```

Comment:
I don't think that this async operation should keep the text controller alive. If the user closes all the text editors, the controller should also finish as soon as possible, and any ongoing operations should be abandoned.

Suggestion is:
1) get the future from `controller.read_content()` outside the async block, then you can await on it without requiring controller
2) pass weak handle to the controller and try upgrading it and writing, after future is done